handlers/citations.go

```diff
@@ -0,0 +1,68 @@
+package handlers
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+
+	ml "github.com/KingDerp/memoryLane"
+	"github.com/KingDerp/memoryLane/server"
+
+	"github.com/sirupsen/logrus"
+)
+
+type citationHandler struct {
+	citationServer *server.CitationServer
+}
+
+func newCitationHandler(server *server.CitationServer) *citationHandler {
+	return &citationHandler{citationServer: server}
+}
+
+func (ss *citationHandler) newCitation(w http.ResponseWriter, req *http.Request) {
+
+	ctx := req.Context()
+
+	newCitationRequest, err := unmarshalCitationReq(req.Body)
+	if err != nil {
+		ml.HandleUnmarshalError(w, err)
```

Comment:
can all of these `err != nil` cases be handled the same with `HandleCitationError` since it will properly handle the various errors? future refactors could cause errors other than marshal/unmarshal errors to be returned